examples/java/core/nothing_default/src/main/java/com/microsoft/bond/examples/NothingDefault.java

```diff
@@ -0,0 +1,87 @@
+package com.microsoft.bond.examples;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.LinkedList;
+
+import com.microsoft.bond.BondSerializable;
+import com.microsoft.bond.Deserializer;
+import com.microsoft.bond.Serializer;
+import com.microsoft.bond.Something;
+import com.microsoft.bond.StructBondType;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+
+import com.microsoft.bond.examples.nothingdefault.Struct_v1;
+import com.microsoft.bond.examples.nothingdefault.Struct_v2;
+
+public class NothingDefault {
+
+    public static void main(final String[] args) throws IOException {
+
+        final Struct_v1 v1 = new Struct_v1();
+
+        boolean sawException = false;
+
+        // Struct_v1 has a required field foo which by default is set to 'nothing'.
+        // If we try to serialize object v1 w/o initializing the field to some value
+        // Bond will throw an exception.
+        try {
+            System.out.println("Serializing v1...");
+            serialize(v1);
```

Comment:
This has been fixed. Fix is in core review.

---
In reply to: [133536169](https://github.com/Microsoft/bond/pull/566#discussion_r133536169) [](ancestors = 133536169)